packages/ember-views/lib/views/states/has_element.js

```diff
@@ -15,8 +16,21 @@ var hasElement = Object.create(_default);
 
 assign(hasElement, {
   $(view, sel) {
-    var elem = view.element;
-    return sel ? jQuery(sel, elem) : jQuery(elem);
+    if (isEnabled('ember-views-tagless-jquery') && view.tagName === '') {
+      var firstNode = view._renderNode.firstNode;
+      var lastNode = view._renderNode.lastNode;
+      var range = jQuery(firstNode).nextUntil(lastNode);
+
+      //Test if we should include first and last nodes in the range.
+      if (firstNode.nodeType !== 8) { range = range.add(firstNode); }
+      if (lastNode.nodeType !== 8) { range = range.add(lastNode); }
```

Comment:
From what I understood, if the tagless component does not contain any DOM, `firstNode` and `lastNode` are HTML comments `<!-- -->`.

Without these checks, calling `this.$()` in a tagless component without any DOM would return a jquery object with the HTML component and length 1.

This make sure we don't add those HTML comments.

This tests assure that the length of the jquery object returned is still zero on such cases: https://github.com/emberjs/ember.js/pull/12500/files#diff-e5717855880c51e81be1741ebbc44c2aR71